test/unittest/unittest_nntrainer_layers.cpp

```diff
@@ -60,23 +60,12 @@ class nntrainer_abstractLayer : public ::testing::Test {
     in = nntrainer::Tensor(layer.getInputDimension()[0]);
     out = nntrainer::Tensor(layer.getOutputDimension()[0]);
 
-    layer.resizeNetInput(layer.getNumInputs());
-    layer.resizeNetOutput(layer.getNumOutputs());
-
-    for (unsigned int i = 0; i < layer.getNumInputs(); ++i) {
-      std::shared_ptr<nntrainer::NetBuffers> n_buffer =
-        std::make_unique<nntrainer::NetBuffers>();
-      n_buffer->var = nntrainer::Tensor(layer.getInputDimension()[i]);
-      layer.setInputBuffer(i, n_buffer);
-    }
-
-    for (unsigned int i = 0; i < layer.getNumOutputs(); ++i) {
-      std::shared_ptr<nntrainer::NetBuffers> n_buffer =
-        std::make_unique<nntrainer::NetBuffers>();
-      n_buffer->var = nntrainer::Tensor(layer.getOutputDimension()[i]);
-      layer.setOutputBuffer(i, n_buffer);
-    }
+    manager.TrackLayerInOuts(layer.getName(), layer.getInputDimension());
+    layer.setInputBuffers(manager.getInputsLayer(-1));
+    manager.TrackLayerInOuts(layer.getName(), layer.getOutputDimension());
```

Comment:
How about `TrackLayerInOuts` return a reference to `in_outs.back()`? Would it be considered as an anti pattern?

Reply:
this is being done in a upcoming PR.

Reply:
👍